### ISSUE TITLE
Updating project.conf with a new url does not cause an update of the ostree config

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#217]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/217)
In GitLab by [[Gitlab user @benjamb]](https://gitlab.com/benjamb) on Jan 26, 2018, 16:05

I initially changed the contents of buildstream.conf to update the push url to `1.2.3.4` from `4.3.2.1`, however, I forgot to update the contents of the project.conf. Resulting in the config of my ostree cache looking like:
```
~/.cache/buildstream/artifacts/ostree$ cat config
[core]
repo_version=1
mode=bare-user

[remote "ssh___push_1_2_3_4_artifacts_"]
url=http://4.3.2.1/artifacts/
gpg-verify=false
```
however, after updating my project.conf with:
```
artifacts:
  url: http://1.2.3.4/artifacts/
```
the ostree config did not update the url of the remote.

[[Gitlab user @samthursfield]](https://gitlab.com/samthursfield) Took a quick look and pointed out https://github.com/ostreedev/ostree/blob/master/src/libostree/ostree-repo.h#L201, essentially, if a remote exists with that name, it won't do anything.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @benjamb]](https://gitlab.com/benjamb) on Jan 26, 2018, 16:07

<     ssam2> i think we'll have to explicitly check if it exists, and if so update the URL in a separate call  
<     ssam2> which looks annoying actually  
<     ssam2> probably need to call .get_config(), then set the 'url' key, then call .write_config()

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @adds68]](https://gitlab.com/adds68) on Jan 29, 2018, 10:28

There is currently a PR open to get the OSTree API updated, to allow updates:

https://github.com/ostreedev/ostree/pull/1166

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 08:30

mentioned in commit a615e00185b097a3ff3ce15f58e79a803339d9f8

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 08:51

mentioned in commit 76cbcd4ec52c26fc77134533df64c64221610e22

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 17:06

mentioned in commit 8c7ab2f0a3582cde19d5be7f293b47bfa51efbc4

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 10, 2018, 18:12

mentioned in commit 1792150b63971e2c4335f785cb9c7da1243cfcc4

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 16, 2018, 15:33

mentioned in commit cb79b711f0c653c10d78e44f3154eee4f9d123b9

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 16, 2018, 17:25

mentioned in commit fdbc1e33f84fd45f6ac4527846fc3cc8a14d009b

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 17, 2018, 06:58

mentioned in commit 1f8b4aa290a908a697f008a29ea143a9320dd639

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 17, 2018, 08:59

closed via merge request !337

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @juergbi]](https://gitlab.com/juergbi) on Jul 17, 2018, 08:59

closed via commit 1f8b4aa290a908a697f008a29ea143a9320dd639
